db/migrate/20230427141829_create_submissions.rb

```diff
@@ -0,0 +1,28 @@
+class CreateSubmissions < ActiveRecord::Migration[7.0]
+  # JSON versus JSONB, and `GIN` (Generalized Inverted Index) indexes
+  # https://www.dbvis.com/thetable/json-vs-jsonb-in-postgresql-a-complete-comparison/
+  # https://nandovieira.com/using-postgresql-and-jsonb-with-ruby-on-rails
+
+  def change
+    create_table :submissions, id: :uuid do |t|
+      t.references :bulk_submission, type: :uuid, foreign_key: true, index: true
+
+      t.datetime :period_start_at
+      t.datetime :period_end_at
+      t.string :use_case, null: false
+      t.string :first_name, null: false, default: ""
+      t.string :last_name, null: false, default: ""
+      t.datetime :dob
+      t.string :nino, null: false, default: ""
+      t.string :status, null: false, default: ""
+      t.uuid :hmrc_interface_id
+      t.jsonb :hmrc_interface_result, null: false, default: '{}'
```

Comment:
I was not familiar with the jsonb data type, but have just read that it can be quicker/more efficient 👍 I had a quick google and found this https://www.educba.com/postgresql-json-vs-jsonnb/ which lists some of the pros and cons - one question I had was whether we ever had any duplicate key/value pairs in hmrc responses as it appears these are not preserved by jsonb - have you tested at all storing actual hmrc responses in the hmrc_interface_result column and if so were there any issues?

Reply:
I have not tested no. If we do find problems then we can use either `json` or even an attachment. duplicate key value pairs at the same nesting level should be invalid really, but who knows what HMR will send back 🤷‍♂️